Update docker stats collector for ecs

dfuentes opened the pull request:
- ECS mode replaces container id with the first 8 digits of the ECS task
  arn, and also pulls the container name from the 'tag' docker
  label (and splits on double dashes).
- This also slightly changes the metric name to add 'docker' before the
  specific measurement.  This is for ease of matching via the influx
  graphite plugin.

Commit:
Add ECS-Mode option to docker_stats collector:

- ECS mode replaces container id with the first 8 digits of the ECS task
  arn, and also pulls the container name from the 'tag' docker
  label (and splits on double dashes).
- This also slightly changes the metric name to add 'docker' before the
  specific measurement.  This is for ease of matching via the influx
  graphite plugin.

Commit:
update tests for docker_stats collector

mohit commented on src/collectors/docker_stats/docker_stats.py:
```diff
@@ -61,14 +64,25 @@ def collect(self):
+            idlabel = parts[1][:8]
+            name = sanitize_delim(tag, "--")
+
+
```
extra line?

mohit replied:
still extra line

dfuentes replied:
1f380fa

mohit commented on src/collectors/docker_stats/docker_stats.py:
```diff
@@ -61,14 +64,25 @@ def collect(self):
+          name = sanitize_delim(name, "/")
+        if self.config['ecs_mode']:
+          labels = container['Config']['Labels']
+          tag = labels.get('tag', '')
```
should this be using the `name_from_env` instead of the hardcoded `tag`?

mohit commented:
style nits. otherwise lgtm

Commit:
add support for multiple networks:

- Docker recently added support for multiple network interfaces to be
  output in docker stats.  This change accounts for that while remaining
  backwards compatible with the old schema.

Commit:
update tests for multiple networks per container

Commit:
fix typo

Commit:
sometimes there is no network in stats block

dfuentes commented:
ping @mohit, had to add a few more changes to get it working in dev

Commit:
remove extra line

mohit commented on src/collectors/docker_stats/docker_stats.py:
```diff
@@ -96,9 +110,16 @@ def collect(self):
+        if not networks:
+          single_network = stats.get('network', {})
+          if single_network:
+            networks = {'eth0': stats['network']}
```
use `single_network` here?

mohit commented:
💯 on tests!
